ggml/src/ggml-vulkan/vulkan-shaders/vulkan-shaders-gen.cpp

```diff
@@ -198,10 +198,15 @@ static uint32_t compile_count = 0;
 static std::mutex compile_count_mutex;
 static std::condition_variable compile_count_cond;
 
-void string_to_spv_func(const std::string& _name, const std::string& in_fname, const std::map<std::string, std::string>& defines, bool fp16 = true, bool coopmat = false, bool coopmat2 = false, bool f16acc = false) {
+void string_to_spv_func(const std::string& _name, const std::string& in_fname, const std::map<std::string, std::string>& defines, bool fp16 = true, bool coopmat = false, bool coopmat2 = false, bool f16acc = false, bool is_embed = false) {
     std::string name = _name + (f16acc ? "_f16acc" : "") + (coopmat ? "_coopmat" : "") + (coopmat2 ? "_cm2" : (fp16 ? "" : "_fp32"));
     std::string out_fname = join_paths(output_dir, name + ".spv");
-    std::string in_path = join_paths(input_dir, in_fname);
+    std::string in_path;
+    if (is_embed) {
+        in_path = join_paths(input_dir + "/../../ggml-kompute/kompute-shaders", in_fname);
```

Comment:
I think it's better to copy shaders that we want to use, to keep the backends separated. This also means that we can do some Vulkan-backend-specific changes or optimizations.

Reply:
OK. That's why I did at first, but felt weird having duplicated files. But if you prefer that option, I'm fine with it.